tests/MongoDB.Driver.Legacy.Tests/Builders/IndexKeysBuilderTests.cs

```diff
@@ -219,5 +254,25 @@ public void TestTextIndexCreation()
             Assert.Equal("spanish", index["default_language"].AsString);
             Assert.Equal(1, index["key"]["c"].AsInt32);
         }
+
+        [Fact]
+        public void Wildcard_should_return_expected_result()
+        {
+            var keys = IndexKeys.Wildcard();
+            BsonDocument expected = BsonDocument.Parse("{ \"$**\" : 1 }");
+            keys.ToBsonDocument().Should().Be(expected);
+
+            keys = IndexKeys.Wildcard("b");
```

Comment:
Should this even be allowed? Or should we require "b.$**"?

What is the purpose of providing two ways to do the same thing. More than one way to do the same thing can cause confusion.

Reply:
The idea here was to hide the mongo syntax under the `wildcard` c# method. So, if I don't know the syntax with `$**`, I can just specify which property I want to use as wildcard index. However, If I know the syntax and expect to set `$**` explicitly, I will still be able to do it.

>can cause confusion.

It also can be described as flexibility (I know it's a word game :) ) if you think that it's more confusing than flexibility, I can remove one from ways, probably, it will be with explicitly specifying `$**`.

Reply:
Having two ways to do things is rarely advantageous. It's better to have one optimal way to do it.

I defer to @jyemin  on this.

Reply:
It seems consistent with other methods in this class to hide the `$**` syntax, so if I have to choose I would go with `IndexKey.Wildcard("b")`, and additionally:

* document what the driver generates, so it's clear to users not to use the `$**` suffix
* check for the `$**` suffix and throw if it's included in the parameter

Reply:
Sorry for being late to the party 😃 Was a decision made offline about allowing the `$**` suffix?

Reply:
The decision here that if the `Wildcard` method doesn't have a parameter, we will add `$**`, otherwise - we will add `parameter` + `.$**`. We rely on the server error in case of any possible wrong queries.

Reply:
Sgtm! Thanks for clarifying!